src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Services/DataSetVersionService.cs

```diff
@@ -23,31 +23,50 @@ internal class DataSetVersionService(
     IDataSetVersionPathResolver dataSetVersionPathResolver
 ) : IDataSetVersionService
 {
+    public async Task<Either<ActionResult, (Guid dataSetId, Guid dataSetVersionId)>> CreateInitialVersion(
+        Guid dataSetId,
+        Guid releaseFileId,
+        Guid instanceId,
+        CancellationToken cancellationToken = default)
+    {
+        return await GetDataSet(dataSetId, cancellationToken)
+            .OnSuccess(ValidateInitialDataSet)
+            .OnSuccess(dataSet => CreateDataSetVersion(
+                releaseFileId: releaseFileId,
+                instanceId: instanceId,
+                dataSet: dataSet, 
+                cancellationToken: cancellationToken));
+    }
+
+    public async Task<Either<ActionResult, (Guid dataSetId, Guid dataSetVersionId)>> CreateNextVersion(
```

Comment:
Same as above - no need to return the `dataSetId` in a tuple when we already know what the `dataSetId` is in the consumer.

Would simplify this by changing the return type to `Task<Either<ActionResult, Guid>>`